### ISSUE TITLE
Flaw in '0016_series_project' migration

### ISSUE THREAD

**Issue by stephenfin** [getpatchwork/patchwork#104]
_From [Mailing List](https://lists.ozlabs.org/pipermail/patchwork/2017-June/004373.html)_

I'm trying to upgrade an older patchwork instance I have, and it seems to be failing when I do a 'migrate' with the specific error:

    django.db.utils.OperationalError: cannot ALTER TABLE "patchwork_series" because it has pending trigger events

I've made sure to update `django-filter` and `django-filters` to be up to date, along with Django. It's using postgres as the back-end (`psycopg2` is the provider). Error logs are below.

---

```
[pwuser@mail patchwork]$ python3 ./manage.py migrate
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, patchwork, sessions, sites
Running migrations:
  Applying patchwork.0016_series_project...Traceback (most recent call last):
  File "/usr/lib/python3.4/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
psycopg2.OperationalError: cannot ALTER TABLE "patchwork_series" because it has pending trigger events


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "./manage.py", line 11, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/lib/python3.4/site-packages/django/core/management/__init__.py", line 363, in execute_from_command_line
    utility.execute()
  File "/usr/lib/python3.4/site-packages/django/core/management/__init__.py", line 355, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/lib/python3.4/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/lib/python3.4/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/usr/lib/python3.4/site-packages/django/core/management/commands/migrate.py", line 204, in handle
    fake_initial=fake_initial,
  File "/usr/lib/python3.4/site-packages/django/db/migrations/executor.py", line 115, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/usr/lib/python3.4/site-packages/django/db/migrations/executor.py", line 145, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/usr/lib/python3.4/site-packages/django/db/migrations/executor.py", line 244, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/usr/lib/python3.4/site-packages/django/db/backends/base/schema.py", line 93, in __exit__
    self.execute(sql)
  File "/usr/lib/python3.4/site-packages/django/db/backends/base/schema.py", line 120, in execute
    cursor.execute(sql, params)
  File "/usr/lib/python3.4/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
  File "/usr/lib/python3.4/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/usr/lib/python3.4/site-packages/django/utils/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/usr/lib/python3.4/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
django.db.utils.OperationalError: cannot ALTER TABLE "patchwork_series" because it has pending trigger events
```

**Comment by stephenfin** [comment]
I'm thinking this happens because this particular migration couples a data migration with a schema migration. [Random StackOverflow questions](https://stackoverflow.com/q/12838111/) suggest this shouldn't happen.

**Comment by orgcandman** [comment]
As confirmed on the mailing list - the patch submitted at http://patchwork.ozlabs.org/patch/775025/ works for me.

**Comment by stephenfin** [comment]
Thanks, @orgcandman. This is resolved in `v2.0.0-rc4`.
